test_helpers/response.go

```diff
@@ -0,0 +1,78 @@
+package test_helpers
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/vmihailenco/msgpack/v5"
+
+	"github.com/tarantool/go-tarantool/v2"
+)
+
+// MockResponse is a mock response used for testing purposes.
+type MockResponse struct {
+	// header contains response header
+	header tarantool.Header
+	// data contains data inside a response.
+	data []byte
+}
+
+// GetData is a getter for a data field.
+func (resp *MockResponse) GetData() []byte {
+	return resp.data
+}
```

Comment:
It is used to get serialized data from the `MockResponse`.

For example, if I want to set a response for the future, I will use
```go
resp := test_helpers.NewMockResponse(t, []interface{}{'v', '2'})

fut := NewFuture(test_helpers.NewMockRequest())
fut.SetResponse(Header{}, bytes.NewBuffer(resp.GetData()))
```
And I will set a correct response body, that will be correctly de-serialized in the `Decode`.

Reply:
I suppose it would be more clear to test `Future` in that way:

```
fut := NewFuture(test_helpers.NewMockRequest())
fut.SetResponse(Header{}, test_helpers.SomeHelpFunctionWithProperName([]interface{}{'v', '2'}))
```

Reply:
Or even write a custom `Request` mock for the `Future` test because you need to test a custom scenarios here:

1. `SetResponse()` calls a create response method from a `Request`.
2. `Get()` calls `Decode()` from a created response and just pass results.
3. `GetTyped()` calls `DecodeTyped()` from a created response and just pass results.
4. `Get()/GetTyped()` saves an error and don't call `Decode()/DecodeTyped()` after an error.
5. `Get()` saves a results and don't call `Decode()` after a result.
6. `GetTyped()` doesn't save a result and call `DecodeTyped()` on each call.